### DISCUSSION TITLE
Can typst be installed on Ubuntu?

### DISCUSSION THREAD

**Discussion by andres-gimenez** [typst/typst#4081] [Q&A] [answered]
Can typst be installed on Ubuntu with **apt install typst**? If not, will it be possible in the future?

**Comment by alerque** [comment, marked as answer]
See #3679. Ubuntu packaging is typically downstream of Debian. There might be 3rd party APT repos out there you can tie into already, but as far as official distro packages they are going to be sometime after it lands in Debian. Also not that Ubuntu's release cycles are quite long and quite frequently struggle with having up to date versions of things. Even if you use it as a base system for ... reasons ... do consider other ways of installing and using fresher versions of specific tools you actually use in your projects.

**Comment by leggewie** [comment]
[install it via snap](https://snapcraft.io/install/typst/ubuntu)

**Reply by InAnYan** [reply]
Don't install it via snap. It's too restrictive on local files...

See also https://discordapp.com/channels/1054443721975922748/1287474369681031299/1287474369681031299 and It's currently to restrictive, there is an open issue <https://github.com/typst/typst-snap/issues/2>

**Comment by cardiosMircem** [comment]
I use Ubuntu on a regular basis and this is how I did it.

### Typst installation
What it worked for me was downloading the [typst-x86_64-unknown-linux-musl.tar.xz](https://github.com/typst/typst/releases/download/v0.12.0/typst-x86_64-unknown-linux-musl.tar.xz) from https://github.com/typst/typst/releases/tag/v0.12.0, decompressed the folder with
```tar -xvf typst-x86_64-unknown-linux-musl.tar.xz```
at this point I get into the decompressed folder and launch the command
```./typst```
via terminal and that's it. I installed tynimist Typst in vscode extension and it works like a charm.

### Package installation
So far I did need just the [fontawesome package](https://typst.app/universe/package/fontawesome/) so I downloaded the desktop version from [here](https://fontawesome.com/download) then I navigate to the otfs folder and installed them one by one by opening each of them and clicking install as in the picture
![Screenshot from 2025-01-28 09-38-21](https://github.com/user-attachments/assets/e0511958-08ff-4561-a2f5-bc33ea82d584)

after that I pasted
```#import "@preview/fontawesome:0.5.0"``` 
in my typst file and I saw the icons. 
However, it seems that the otf file `Font Awesome 6 Free-Solid-900` does not get installed but I can use the icons anyway so I am ok with this
